caddyhttp/httpserver/server.go

```diff
@@ -507,16 +507,44 @@ func (s *Server) Stop() error {
 // OnStartupComplete lists the sites served by this server
 // and any relevant information, assuming caddy.Quiet == false.
 func (s *Server) OnStartupComplete() {
+
 	if caddy.Quiet {
 		return
 	}
+
+	s.OutputSiteInfo()
+
+}
+
+func (s *Server) OutputSiteInfo() {
+
+	sitesByPort := make(map[string][]SiteConfig)
 	for _, site := range s.sites {
-		output := site.Addr.String()
-		if caddy.IsLoopback(s.Address()) && !caddy.IsLoopback(site.Addr.Host) {
-			output += " (only accessible on this machine)"
+		// Group Sites together by port before outputing
+		sitesArray := sitesByPort["port "+site.Port()]
+		sitesArray = append(sitesArray, *site)
+		sitesByPort["port "+site.Port()] = sitesArray
+	}
+
+	fmt.Println(" ")
+	for key, asiteType := range sitesByPort {
+		site := asiteType[0]
+		scheme := "HTTP"
+		if site.TLS.Enabled {
+			scheme = "HTTPS"
+		}
+
+		fmt.Printf("Serving %s on "+key+" \n", scheme)
+
+		for _, site = range asiteType {
+			output := site.Addr.String()
+			if caddy.IsLoopback(s.Address()) && !caddy.IsLoopback(site.Addr.Host) {
+				output += " (only accessible on this machine)"
+			}
+			fmt.Println(output)
+			log.Println(output)
```

Comment:
I think the log does not need this "grouped" output so much, it seems to add a bit of noise. Perhaps we should rethink how we log this instead. Maybe like what we had before, but instead, it'd look something like `[INFO] Serving %s` (site address)

Reply:
Just to clarify @mholt, you think that the outline in the PR description - i.e.

```
Serving HTTP on port 80
http://example.com

Serving HTTPS on port 443
https://example.com
```

isn't best?

Reply:
Correct, for log output. Logs should have the relevant information and context on a single line when possible.